File: frontend/src/types/smartReply.ts
/**
 * Типы для smart replies
 */

export interface SmartReplyRequest {
    message_id: string;
    context_messages?: number;
}

export interface SmartReplyResponse {
    replies: string[];
    confidence?: number[];
}

export interface SmartReplySettings {
    enabled: boolean;
    max_suggestions: number;
    include_emoji: boolean;
    formal_style: boolean;
}

export const DEFAULT_SMART_REPLY_SETTINGS: SmartReplySettings = {
    enabled: true,
    max_suggestions: 3,
    include_emoji: true,
    formal_style: false,
};

File: backend/src/schemas/smart_reply.py
"""Схемы для smart replies."""

import uuid

from pydantic import BaseModel, Field


class SmartReplyRequest(BaseModel):
    """Запрос на генерацию умных ответов."""

    message_id: uuid.UUID
    context_messages: int = Field(5, ge=1, le=20)


class SmartReplyResponse(BaseModel):
    """Ответ с вариантами умных ответов."""

    replies: list[str]
    confidence: list[float] | None = None


class SmartReplySettingsRequest(BaseModel):
    """Настройки smart replies."""

    enabled: bool = True
    max_suggestions: int = Field(3, ge=1, le=5)
    include_emoji: bool = True
    formal_style: bool = False


class SmartReplySettingsResponse(BaseModel):
    """Текущие настройки smart replies."""

    enabled: bool
    max_suggestions: int
    include_emoji: bool
    formal_style: bool


QUICK_REPLIES = {
    "greeting": [
        "Привет!",
        "Здравствуй!",
        "Добрый день!",
    ],
    "thanks": [
        "Спасибо!",
        "Благодарю!",
        "Отлично, спасибо!",
    ],
    "agreement": [
        "Хорошо",
        "Договорились",
        "Отлично!",
    ],
    "question": [
        "Да",
        "Нет",
        "Возможно",
    ],
    "farewell": [
        "Пока!",
        "До свидания!",
        "До связи!",
    ],
    "confirmation": [
        "Понял",
        "Принято",
        "Ок, сделаю",
    ],
}
